diff -up authconfig-6.2.8/authinfo.py.initgroups authconfig-6.2.8/authinfo.py
--- authconfig-6.2.8/authinfo.py.initgroups	2016-06-29 18:01:35.000000000 +0200
+++ authconfig-6.2.8/authinfo.py	2016-08-31 12:31:26.880677634 +0200
@@ -2030,6 +2030,7 @@ class AuthInfo:
 		# Open the file.  Bail if it's not there or there's some problem
 		# reading it.
 		nssconfig = ""
+		initgroups = ""
 
 		try:
 			f = open(all_configs[CFG_NSSWITCH].origPath, "r")
@@ -2038,7 +2039,7 @@ class AuthInfo:
 
 		for line in f:
 			line = line.strip()
-			
+
 			value = matchKey(line, "passwd:")
 			if value:
 				nssconfig = value
@@ -2059,6 +2060,10 @@ class AuthInfo:
 					dnspos = checkNSS(value, "dns")
 					if nispos != None and dnspos != None:
 						self.setParam("preferDNSinHosts", dnspos < nispos, ref)
+				else:
+					value = matchKey(line, "initgroups:")
+					if value:
+						initgroups = value
 
 		if nssconfig:
 			nssmap = (('Compat', 'compat'), ('DB', 'db'),
@@ -2068,6 +2073,8 @@ class AuthInfo:
 			for attr, nssentry in nssmap:
 				if checkNSS(nssconfig, nssentry):
 					self.setParam('enable' + attr, True, ref)
+			if initgroups and checkNSS(nssconfig, 'sss') and not checkNSS(initgroups, 'sss'):
+				self.inconsistentAttrs.append('implicitSSSD')
 
 			self.setParam("implicitSSSD", bool(checkNSS(nssconfig, "sss")), ref)
 		f.close()
