diff -Naur old/authconfig.spec.in new/authconfig.spec.in
--- old/authconfig.spec.in	2013-11-01 16:08:01.000000000 +0100
+++ new/authconfig.spec.in	2017-03-28 13:31:57.676934533 +0200
@@ -9,7 +9,7 @@
 Source: https://fedorahosted.org/releases/a/u/%{name}/%{name}-%{version}.tar.bz2
 Requires: newt-python, pam >= 0.99.10.0, python, libpwquality > 0.9
 Conflicts: pam_krb5 < 1.49, samba-common < 3.0, samba-client < 3.0
-Conflicts: nss_ldap < 254, sssd < 0.99.1
+Conflicts: nss_ldap < 254, sssd < 1.15.1
 Conflicts: freeipa-client < 2.2.0, ipa-client < 2.2.0
 BuildRequires: glib2-devel, python >= 2.6, python-devel
 BuildRequires: desktop-file-utils, intltool, gettext, perl-XML-Parser
diff -Naur old/authinfo.py new/authinfo.py
--- old/authinfo.py	2017-03-28 13:30:42.159279958 +0200
+++ new/authinfo.py	2017-03-28 13:34:02.955701552 +0200
@@ -3827,7 +3827,7 @@
 			# do not continue to following modules if authentication fails
 			if name == "unix" and stack == "auth" and (self.enableSSSDAuth or
 				self.implicitSSSDAuth or self.enableIPAv2) and (not self.enableNIS):
-				logic = LOGIC_FORCE_PKCS11 # make it or break it logic
+				logic = LOGIC_SUFFICIENT
 			# use oddjob_mkhomedir if available
 			if name == "mkhomedir" and os.access("%s/pam_%s.so"
 				% (AUTH_MODULE_DIR, "oddjob_mkhomedir"), os.X_OK):
