diff -up authconfig-6.1.12/authinfo.py.ldapbase authconfig-6.1.12/authinfo.py
--- authconfig-6.1.12/authinfo.py.ldapbase	2014-06-17 15:54:07.000000000 +0200
+++ authconfig-6.1.12/authinfo.py	2014-06-17 16:06:07.851303543 +0200
@@ -185,6 +185,22 @@ def stringsDiffer(a, b, case_sensitive):
 	else:
 		return a.lower() != b.lower()
 
+# Heuristic check whether a string is LDAP DN
+def checkDN(value):
+	lst = value.lstrip().split("=",1)
+	if len(lst) != 2:
+		return False
+	if " " in lst[0]:
+		return False
+	return True
+
+def matchBaseLine(line, key):
+	value = matchKey(line, key)
+	if value:
+		return checkDN(value)
+	else:
+		return False
+
 # Check for a string in an nss configuration line.
 def checkNSS(configuration, candidate):
 	lst = configuration.split(":",1)
@@ -1475,7 +1491,7 @@ class AuthInfo:
 
 			# Is it a "base" statement?
 			value = matchKey(line, "base")
-			if value:
+			if value and checkDN(value):
 				# Save the base DN.
 				self.setParam("ldapBaseDN", value, ref)
 				continue
@@ -2375,7 +2391,7 @@ class AuthInfo:
 				elif matchLine(ls, host):
 					if self.ldapServer:
 						output += "#" + line 
-				elif matchLine(ls, base):
+				elif matchBaseLine(ls, base):
 					# If it's a 'base' line, insert ours instead.
 					if not wrotebasedn and self.ldapBaseDN:
 						output += base + " "
