diff -up authconfig-6.2.8/authconfig-gtk.py.localetb authconfig-6.2.8/authconfig-gtk.py
--- authconfig-6.2.8/authconfig-gtk.py.localetb	2015-07-03 10:59:22.000000000 +0200
+++ authconfig-6.2.8/authconfig-gtk.py	2015-07-03 11:14:22.666532393 +0200
@@ -30,7 +30,10 @@ _ = gettext.lgettext
 import locale
 import dbus
 
-locale.setlocale(locale.LC_ALL, '')
+try:
+	locale.setlocale(locale.LC_ALL, '')
+except locale.Error:
+	sys.stderr.write('Warning: Unsupported locale setting.\n')
 
 firstbootservices = [
 	"autofs",
diff -up authconfig-6.2.8/authconfig.py.localetb authconfig-6.2.8/authconfig.py
--- authconfig-6.2.8/authconfig.py.localetb	2015-07-03 10:59:22.000000000 +0200
+++ authconfig-6.2.8/authconfig.py	2015-07-03 11:14:01.694065135 +0200
@@ -28,7 +28,11 @@ import gettext, os, signal, sys
 _ = gettext.lgettext
 from optparse import OptionParser, IndentedHelpFormatter
 import locale
-locale.setlocale(locale.LC_ALL, '')
+
+try:
+	locale.setlocale(locale.LC_ALL, '')
+except locale.Error:
+	sys.stderr.write('Warning: Unsupported locale setting.\n')
 
 def runsAs(name):
 	return sys.argv[0].find(name) >= 0
