diff -up authconfig-6.2.8/authconfig-gtk.py.winbind-client authconfig-6.2.8/authconfig-gtk.py
--- authconfig-6.2.8/authconfig-gtk.py.winbind-client	2014-09-29 15:19:15.000000000 +0200
+++ authconfig-6.2.8/authconfig-gtk.py	2014-09-29 15:37:19.586350537 +0200
@@ -130,7 +130,7 @@ class Authconfig:
 			 "nisoptions", "nis_map", authinfo.PATH_LIBNSS_NIS, "ypbind"),
 			"Winbind":
 			(_("Winbind"), ("WinbindAuth",),
-			 "winbindoptions", "winbind_map", authinfo.PATH_WINBIND, "samba-winbind-clients"),
+			 "winbindoptions", "winbind_map", authinfo.PATH_WINBIND, "samba-winbind"),
 			"IPAv2":
 			(_("IPAv2"), ("IPAv2Auth",),
 			 "ipav2options", "ipav2_map", authinfo.PATH_IPA_CLIENT_INSTALL, "freeipa-client")
diff -up authconfig-6.2.8/authconfig.py.winbind-client authconfig-6.2.8/authconfig.py
--- authconfig-6.2.8/authconfig.py.winbind-client	2015-07-03 11:34:59.798789958 +0200
+++ authconfig-6.2.8/authconfig.py	2015-07-03 11:33:00.363034242 +0200
@@ -704,8 +704,8 @@ class AuthconfigTUI(Authconfig):
 		warnNIS = [authinfo.PATH_YPBIND, _("NIS"), "ypbind", None]
 		warnShadow = [authinfo.PATH_PWCONV, _("shadow password"), "shadow-utils", None]
 		warnWinbindNet = [authinfo.PATH_WINBIND_NET, _("Winbind"), "samba-client", None]
-		warnWinbindAuth = [authinfo.PATH_PAM_WINBIND, _("Winbind authentication"), "samba-common", warnWinbindNet]
-		warnWinbind = [authinfo.PATH_LIBNSS_WINBIND, _("Winbind"), "samba-common", warnWinbindAuth]
+		warnWinbindAuth = [authinfo.PATH_PAM_WINBIND, _("Winbind authentication"), "samba-winbind", warnWinbindNet]
+		warnWinbind = [authinfo.PATH_LIBNSS_WINBIND, _("Winbind"), "samba-winbind", warnWinbindAuth]
 
 		# Information
 		infoGrid = snack.Grid(1, 6)
