diff -up authconfig-6.2.8/authinfo.py.joinpassword authconfig-6.2.8/authinfo.py
--- authconfig-6.2.8/authinfo.py.joinpassword	2015-07-03 12:47:44.993864700 +0200
+++ authconfig-6.2.8/authinfo.py	2015-07-03 12:53:21.451618592 +0200
@@ -4296,11 +4296,12 @@ class AuthInfo:
 			
 			if echo:
 				sys.stderr.write("[%s]\n" % cmd)
+			if self.joinPassword or not echo:
+				status, error = feedFork(cmd, echo, "sword:", self.joinPassword)
+			else:
 				child = Popen([cmd], shell=True)
 				child.communicate()
 				status = child.returncode
-			else:
-				status, error = feedFork(cmd, echo, "sword:", self.joinPassword)
 			if echo:
 				if status != 0:
 					self.messageCB(_("Winbind domain join was not successful."))
