diff -up authconfig-6.2.8/authinfo.py.no-update authconfig-6.2.8/authinfo.py
--- authconfig-6.2.8/authinfo.py.no-update	2016-06-16 14:27:56.335158214 +0200
+++ authconfig-6.2.8/authinfo.py	2016-06-16 14:35:32.114373306 +0200
@@ -2498,8 +2498,10 @@ class AuthInfo:
 		reallyimplicit = self.sssdSupported()
 		if self.implicitSSSD and not reallyimplicit and not self.enableIPAv2:
 			self.setParam("enableSSSD", True, ref)
+			self.implicitSSSD = False
 		if self.implicitSSSDAuth and not reallyimplicit and not self.enableIPAv2:
 			self.setParam("enableSSSDAuth", True, ref)
+			self.implicitSSSDAuth = False
 
 		self.readLogindefs(ref)
                 self.readPWQuality(ref)
