diff -Naur old/authinfo.py new/authinfo.py
--- old/authinfo.py	2017-05-05 11:47:38.522857750 +0200
+++ new/authinfo.py	2017-05-05 11:57:28.372141053 +0200
@@ -463,6 +463,8 @@
 	 "env",			[]],
 	[True,  AUTH,		LOGIC_REQUIRED,
 	 "faildelay",		argv_faildelay],
+	[False,  AUTH,		LOGIC_REQUIRED,
+	 "faillock",		["preauth", "silent"]],
 	[False,  AUTH,          LOGIC_SKIPNEXT,
 	 "succeed_if",		argv_succeed_if_nonlogin],
 	[False,  AUTH,          LOGIC_PKCS11,
@@ -604,6 +606,8 @@
 	 "env",			[]],
 	[True,  AUTH,		LOGIC_REQUIRED,
 	 "faildelay",		argv_faildelay],
+	[False,  AUTH,		LOGIC_REQUIRED,
+	 "faillock",		["preauth", "silent"]],
 	[False, AUTH,		LOGIC_REQUIRED,
 	 "deny",		[]],
 	[False, AUTH,		LOGIC_SKIPNEXT_ON_FAILURE,
@@ -706,6 +710,8 @@
 	[True,  AUTH,		LOGIC_REQUIRED,
 	 "env",			[]],
 	[False,  AUTH,		LOGIC_REQUIRED,
+	 "faillock",		["preauth", "silent"]],
+	[False,  AUTH,		LOGIC_REQUIRED,
 	 "deny",		[]],
 	[False,  AUTH,		LOGIC_SUFFICIENT,
 	 "fprintd",		[]],
@@ -764,6 +770,8 @@
 pam_modules[SMARTCARD] = [
 	[True,  AUTH,		LOGIC_REQUIRED,
 	 "env",			[]],
+	[False,  AUTH,		LOGIC_REQUIRED,
+	 "faillock",		["preauth", "silent"]],
 	[False,  AUTH,		LOGIC_SUFFICIENT,
 	 "sss",			argv_sssd_missing_name],
 	[False,  AUTH,		LOGIC_PKCS11,
