diff -up authconfig-6.2.8/authinfo.py.altfiles authconfig-6.2.8/authinfo.py
--- authconfig-6.2.8/authinfo.py.altfiles	2014-09-29 15:27:57.000000000 +0200
+++ authconfig-6.2.8/authinfo.py	2014-09-29 15:31:41.371715194 +0200
@@ -1333,6 +1333,9 @@ class AuthInfo:
 		self.preferDNSinHosts = None
 		self.enableSSSD = None
 		self.enableIPAv2 = None
+                # This one we don't have a config entry, we just
+                # preserve the entry if we see it.
+		self.enableAltfiles = None
 
 		# Authentication setup.
 		self.enableAFS = None
@@ -2042,7 +2045,7 @@ class AuthInfo:
 		if nssconfig:
 			nssmap = (('Compat', 'compat'), ('DB', 'db'),
 				  ('Directories', 'directories'), ('Hesiod', 'hesiod'),
-				  ('LDAP', 'ldap'), ('NIS', 'nis'),
+				  ('LDAP', 'ldap'), ('NIS', 'nis'), ('Altfiles', 'altfiles'),
 				  ('NIS3', 'nisplus'), ('Winbind', 'winbind'))
 			for attr, nssentry in nssmap:
 				if checkNSS(nssconfig, nssentry):
@@ -3583,6 +3586,8 @@ class AuthInfo:
 			if self.enableDB:
 				normal += " db"
 			normal += " files"
+                        if self.enableAltfiles:
+                                normal += " altfiles"
 			services = normal
 			if self.enableDirectories:
 				normal += " directories"
