diff -up authconfig-6.2.8/authconfig.py.template-group authconfig-6.2.8/authconfig.py
--- authconfig-6.2.8/authconfig.py.template-group	2016-06-16 11:52:48.081598147 +0200
+++ authconfig-6.2.8/authconfig.py	2016-06-16 14:25:21.353653203 +0200
@@ -233,8 +233,6 @@ class Authconfig:
 			help=_("the character which will be used to separate the domain and user part of winbind-created user names if winbindusedefaultdomain is not enabled"))
 		parser.add_option("--winbindtemplatehomedir", metavar="</home/%D/%U>",
 			help=_("the directory which winbind-created users will have as home directories"))
-		parser.add_option("--winbindtemplateprimarygroup", metavar="<nobody>",
-			help=_("the group which winbind-created users will have as their primary group"))
 		parser.add_option("--winbindtemplateshell", metavar="</bin/false>",
 			help=_("the shell which winbind-created users will have as their login shell"))
 		parser.add_option("--enablewinbindusedefaultdomain", action="store_true",
@@ -490,7 +488,6 @@ class Authconfig:
 			"smbidmaprange":"smbIdmapRange",
 			"winbindseparator":"winbindSeparator",
 			"winbindtemplatehomedir":"winbindTemplateHomedir",
-			"winbindtemplateprimarygroup":"winbindTemplatePrimaryGroup",
 			"winbindtemplateshell":"winbindTemplateShell",
 			"ipav2domain":"ipav2Domain",
 			"ipav2realm":"ipav2Realm",
diff -up authconfig-6.2.8/authinfo.py.template-group authconfig-6.2.8/authinfo.py
--- authconfig-6.2.8/authinfo.py.template-group	2016-06-16 13:53:03.002686209 +0200
+++ authconfig-6.2.8/authinfo.py	2016-06-16 14:25:21.354653226 +0200
@@ -1303,7 +1303,6 @@ class AuthInfo:
 
 		self.winbindSeparator = ""
 		self.winbindTemplateHomedir = ""
-		self.winbindTemplatePrimaryGroup = ""
 		self.winbindTemplateShell = ""
 		self.winbindUseDefaultDomain = None
 		self.winbindOffline = None
@@ -1437,8 +1436,7 @@ class AuthInfo:
 		("enableFprintd", "b"), ("enableSmartcard", "b"), ("forceSmartcard", "b")]),
 	SaveGroup(self.writeWinbind, self.toggleWinbindService, [("smbWorkgroup", "i"), ("smbServers", "i"),
 		("smbRealm", "c"), ("smbSecurity", "i"), ("smbIdmapRange", "i"),
-		("winbindSeparator", "c"), ("winbindTemplateHomedir", "c"),
-		("winbindTemplatePrimaryGroup", "c"), ("winbindTemplateShell", "c"),
+		("winbindSeparator", "c"), ("winbindTemplateHomedir", "c"), ("winbindTemplateShell", "c"),
 		("winbindUseDefaultDomain", "b"), ("winbindOffline", "b"), ("winbindKrb5", "b")]),
 	SaveGroup(self.writeNSS, None, [("enableDB", "b"), ("enableDirectories", "b"), ("enableWinbind", "b"),
 		("enableOdbcbind", "b"), ("enableNIS3", "b"), ("enableNIS", "b"),
@@ -2003,9 +2001,6 @@ class AuthInfo:
 		tmp = self.readWinbindGlobal("template homedir")
 		if tmp:
 			self.setParam("winbindTemplateHomedir", tmp, ref)
-		tmp = self.readWinbindGlobal("template primary group")
-		if tmp:
-			self.setParam("winbindTemplatePrimaryGroup", tmp, ref)
 		tmp = self.readWinbindGlobal("template shell")
 		if tmp:
 			self.setParam("winbindTemplateShell", tmp, ref)
@@ -3472,10 +3467,6 @@ class AuthInfo:
 			output += "   template homedir = "
 			output += self.winbindTemplateHomedir
 			output += "\n"
-		if self.winbindTemplatePrimaryGroup:
-			output += "   template primary group = "
-			output += self.winbindTemplatePrimaryGroup
-			output += "\n"
 		if self.winbindTemplateShell:
 			output += "   template shell = "
 			output += self.winbindTemplateShell
@@ -3518,8 +3509,7 @@ class AuthInfo:
 		all_configs[CFG_SMB].backup(self.backupDir)
 		options = ["workgroup", "password server", "realm", "security",
 			   "domain logons", "domain master",
-			   "idmap uid", "idmap gid", "winbind separator",
-			   "template homedir", "template primary group",
+			   "idmap uid", "idmap gid", "winbind separator", "template homedir",
 			   "template shell", "winbind use default domain",
 			   "winbind offline logon", "kerberos method"]
 		f = None
