diff -up authconfig-6.1.12/authinfo.py.services authconfig-6.1.12/authinfo.py
--- authconfig-6.1.12/authinfo.py.services	2012-03-27 13:52:25.029033945 +0200
+++ authconfig-6.1.12/authinfo.py	2012-03-27 13:57:32.838437966 +0200
@@ -3079,6 +3079,7 @@ class AuthInfo:
 		wrotenetgroup = False
 		wroteautomount = False
 		wrotehosts = False
+		wroteservices = False
 		f = None
 		output = ""
 		all_configs[CFG_NSSWITCH].backup(self.backupDir)
@@ -3091,6 +3092,7 @@ class AuthInfo:
 			if self.enableDB:
 				normal += " db"
 			normal += " files"
+			services = normal
 			if self.enableDirectories:
 				normal += " directories"
 			if self.enableOdbcbind:
@@ -3101,6 +3103,7 @@ class AuthInfo:
 				normal += " nis"
 			if self.enableSSSD or self.implicitSSSD or self.enableIPAv2:
 				normal += " sss"
+				services += " sss"
 			if self.enableLDAPbind:
 				normal += " ldapbind"
 			if self.enableLDAP and not self.implicitSSSD:
@@ -3197,6 +3200,13 @@ class AuthInfo:
 						output += hosts
 						output += "\n"
 						wrotehosts = True
+				# If it's a 'services' line, insert ours instead.
+				elif matchLine(ls, "services:"):
+					if not wroteservices:
+						output += "services:  "
+						output += services
+						output += "\n"
+						wroteservices = True
 				# Otherwise, just copy the current line out.
 				else:
 					output += line
@@ -3226,6 +3236,10 @@ class AuthInfo:
 				output += "hosts:     "
 				output += hosts
 				output += "\n"
+			if not wroteservices:
+				output += "services:  "
+				output += services
+				output += "\n"
 
 			# Write it out and close it.
 			f.rewind()
