diff -up authconfig-6.1.12/authconfig-gtk.py.nox authconfig-6.1.12/authconfig-gtk.py
--- authconfig-6.1.12/authconfig-gtk.py.nox	2011-01-13 17:32:14.000000000 +0100
+++ authconfig-6.1.12/authconfig-gtk.py	2011-03-25 10:24:42.000000000 +0100
@@ -24,7 +24,6 @@
 # Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 #
 import authinfo, acutil
-import msgarea
 import gettext, os, signal, sys
 _ = gettext.lgettext
 import locale
@@ -60,13 +59,12 @@ if "--nox" in sys.argv:
 try:
 	import gtk, gtk.glade
 except RuntimeError, e:
-	if (os.isatty(sys.stdin.fileno()) and
-	    os.isatty(sys.stdout.fileno()) and
-	    os.isatty(sys.stderr.fileno())):
-		os.execv('/usr/bin/authconfig', ['authconfig']+sys.argv[1:])
-		sys.exit(1)
-	else:
-		raise e
+	sys.stderr.write(_("Unable to initialize graphical environment. Most likely cause of failure\n"
+		"is that the tool was not run using a graphical environment. Please either\n"
+		"start your graphical user interface or set your DISPLAY variable.\n"))
+	sys.exit(2)
+
+import msgarea
 
 class Authconfig:
 	def __init__(self):
